src/test/java/org/opensearch/security/auditlog/compliance/RestApiComplianceAuditlogTest.java

```diff
@@ -108,21 +114,24 @@ public void testRestApiRolesDisabledGet() throws Exception {
             .build();
 
         setup(additionalSettings);
-        TestAuditlogImpl.clear();
+        Thread.sleep(1000);
 
         rh.enableHTTPClientSSL = true;
         rh.trustHTTPServerCertificate = true;
         rh.sendAdminCertificate = true;
         rh.keystore = "kirk-keystore.jks";
-        HttpResponse response = rh.executeGetRequest("_opendistro/_security/api/rolesmapping/opendistro_security_all_access?pretty");
-        Thread.sleep(1500);
-        Assert.assertEquals(HttpStatus.SC_OK, response.getStatusCode());
-        Assert.assertTrue(TestAuditlogImpl.messages.size() > 2);
-        Assert.assertTrue(TestAuditlogImpl.sb.toString().contains("audit_request_effective_user"));
-        Assert.assertTrue(TestAuditlogImpl.sb.toString().contains("COMPLIANCE_INTERNAL_CONFIG_READ"));
-        Assert.assertTrue(TestAuditlogImpl.sb.toString().contains("COMPLIANCE_INTERNAL_CONFIG_WRITE"));
-        Assert.assertTrue(TestAuditlogImpl.sb.toString().contains("UPDATE"));
-        validateMsgs(TestAuditlogImpl.messages);
+        final List<AuditMessage> messages = TestAuditlogImpl.doThenWaitForMessages(() -> {
+            HttpResponse response = rh.executeGetRequest("_opendistro/_security/api/rolesmapping/opendistro_security_all_access?pretty");
+            Assert.assertEquals(HttpStatus.SC_OK, response.getStatusCode());
+            try {
+                Thread.sleep(1500);
+            } catch (InterruptedException e) {
+                fail("Received unexpected exception");
+            }
+            Assert.assertTrue(TestAuditlogImpl.sb.toString().contains("audit_request_effective_user"));
+            Assert.assertTrue(TestAuditlogImpl.sb.toString().contains("COMPLIANCE_INTERNAL_CONFIG_READ"));
+        }, 1);
+        validateMsgs(messages);
     }
 
     @Test
```

Comment:
> This keeps the testAutoInit test. That test verifies that when the security index and audit log index are created that there are audit log entries that capture those operations. Its not the greatest test, but I think we should have some assertions around that behavior.

If the test's goal is to detect specific interactions with the audit system during startup, it wouldn't wrap `setup(...)`; it would be built so individual action appeared in the audit log in an expected order.

As the test is, it is prone to side effects from cluster setup, if the number of messages changed from 4->8 I'm not sure there is a compelling reason not to update the test.  We have many richer test cases in this class that better verify behavior that are overlapping with this tests coverage.

Reply:
Added richer assertions. See comment [here](https://github.com/opensearch-project/security/pull/3733#discussion_r1402446523).